app/src/main/java/com/toygar/tasks/viewmodels/TaskListViewModel.kt

```diff
@@ -1,16 +1,33 @@
 package com.toygar.tasks.viewmodels
 
-import android.content.Context
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
 import androidx.lifecycle.LiveData
-import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.toygar.tasks.models.Task
 import com.toygar.tasks.data.LocalDatabase
+import com.toygar.tasks.models.SortableTaskFields
 import com.toygar.tasks.repositories.LocalDbTaskRepository
 import com.toygar.tasks.repositories.TaskRepository
+import kotlinx.coroutines.launch
 
-class TaskListViewModel(context: Context) : ViewModel() {
+class TaskListViewModel(application: Application) : AndroidViewModel(application) {
     private val taskRepository: TaskRepository =
-        LocalDbTaskRepository(LocalDatabase.getInstance(context).taskDao())
+        LocalDbTaskRepository(LocalDatabase.getInstance(getApplication()).taskDao())
 
     val tasks: LiveData<out List<Task>> = taskRepository.tasks
+
```

Comment:
getter kullanmak daha garanti olur her seferinde repo içindeki değişkenin en güncel halini almak için
```
val tasks: LiveData<out List<Task>>
    get() = taskRepository.tasks
```